src/app/modules/submission/submission.middleware.ts

```diff
@@ -0,0 +1,43 @@
+import { RequestHandler } from 'express'
+import { ParamsDictionary } from 'express-serve-static-core'
+
+import { createLoggerWithLabel } from '../../../config/logger'
+import { WithAutoReplyEmailData } from '../../../types'
+
+import * as SubmissionService from './submission.service'
+import { ProcessedFieldResponse } from './submission.types'
+
+const logger = createLoggerWithLabel(module)
+
+export const sendEmailConfirmations: RequestHandler<
+  ParamsDictionary,
+  unknown,
+  { parsedResponses: ProcessedFieldResponse[] }
+> = async (req, res) => {
+  const {
+    form,
+    attachments,
+    autoReplyData,
+    submission,
+  } = req as WithAutoReplyEmailData<typeof req>
+  // Return the reply early to the submitter
+  res.json({
+    message: 'Form submission successful.',
+    submissionId: submission.id,
+  })
```

Comment:
just to be sure, we're still only returning HTTP 200 after admin mail is sent, right?

Reply:
yup, this middleware is after the `sendAdminEmail` middleware